js/modules/k6/experimental/tracing/module.go

```diff
@@ -69,3 +73,63 @@ func (mi *ModuleInstance) newClient(cc goja.ConstructorCall) *goja.Object {
 
 	return rt.ToValue(NewClient(mi.vu, opts)).ToObject(rt)
 }
+
+// InstrumentHTTP instruments the HTTP module with tracing headers.
+//
+// When used in the context of a k6 script, it will automatically replace
+// the imported http module's methods with instrumented ones.
+func (mi *ModuleInstance) instrumentHTTP(options options) {
+	rt := mi.vu.Runtime()
+
+	if mi.Client != nil {
```

Comment:
Should it be protected from data race? Could it be called concurrently? Apparently, nothing blocks the user from using it in the iteration function.

Reply:
I'm not sure about the potential data race; probably @mstoykov might be more able to give a certain answer on this? My current reasoning is the module is instantiated once for each VU; thus this check will access the `Client` instance only once per-vu at a time, hence, no concurrent access? I might be completely off on this though, as I don't have the VU execution model freshly loaded in my brain 🤔 

Your second point actually brings me to think that considering `instrumentHTTP` is a one-way street at the moment, we probably want to limit it to the init context indeed 🤔

Reply:
Indeed, thanks for the pointer 🙏🏻 I had already done it in xk6-redis too, though 😸 I've added a check, and a couple tests to cover that 🤝

Reply:
> the module is instantiated once for each VU

Yes this is true . 

There is no possible race condition as it will require that this is called twice *from* js at the same time.

But by specification js is single threaded so there is no way for that to happen or it will be a bug in something else - like using a js runtime concurrently.